AdamantTests/AddressGeneratorTests.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  AddressGeneratorTests.swift
+//  AdamantTests
+//
+//  Created by Andrey on 11.07.2022.
+//  Copyright © 2022 Adamant. All rights reserved.
+//
+
+import XCTest
+@testable import Adamant
+
+private struct PublicKeyAndAddress {
+    let publicKey: String
+    let address: String
+}
+
+class AddressGeneratorTests: XCTestCase {
+    func test_0() {
+        test(index: 0)
+    }
+    
+    func test_1() {
+        test(index: 1)
+    }
+    
+    func test_2() {
+        test(index: 2)
+    }
+    
+    func test_3() {
+        test(index: 3)
+    }
+    
+    func test_4() {
+        test(index: 4)
+    }
+    
+    func test_5() {
+        test(index: 5)
+    }
+    
+    func test_6() {
+        test(index: 6)
+    }
+    
+    func test_7() {
+        test(index: 7)
+    }
+    
+    func test_8() {
+        test(index: 8)
+    }
+    
+    func test_9() {
+        test(index: 9)
+    }
+}
+
+private func test(index: Int) {
+    let testData = testDataList[index]
+    XCTAssertEqual(
+        testData.address,
+        AddressGenerator.generateAddress(publicKey: testData.publicKey)
+    )
+}
+
+private let testDataList: [PublicKeyAndAddress] = [
```

Comment:
Don't we want test for negative cases, e. g., when bad PublicKey provided?

Reply:
Okay, I will add bad public key tests